build/mixxx.py

```diff
@@ -39,8 +39,8 @@ def __init__(self, target, machine, build, toolchain, available_features=[]):
         if build is None:
             build = 'debug'
 
-        if not build in ['debug', 'release']:
-            raise Exception("invalid build type")
+        if not build in ['debug', 'release', 'none']:
```

Comment:
Right, but if the person who wrote that check didn't realize there were 3 options and did the opposite check then it would select the debug version. The point is it's entirely arbitrary and it does things the original programmer did not explicitly intend. Adding support for a flag-less build is fine but this way of doing it makes it easy to write bugs. 

It's especially concerning if this none option is what will be used by Linux repository maintainers so we need to be precise about what we are doing.